nvtabular/ops/categorify.py

```diff
@@ -320,31 +325,69 @@ def fit(self, columns: ColumnNames, ddf: dd.DataFrame):
                 warnings.warn("Cannot use `search_sorted=True` for pandas-backed data.")
 
         # convert tuples to lists
-        columns = [list(c) if isinstance(c, tuple) else c for c in columns]
-        dsk, key = _category_stats(
-            ddf,
-            FitOptions(
-                columns,
-                [],
-                [],
-                self.out_path,
-                self.freq_threshold,
-                self.tree_width,
-                self.on_host,
-                concat_groups=self.encode_type == "joint",
-                name_sep=self.name_sep,
-                max_size=self.max_size,
-                num_buckets=self.num_buckets,
-            ),
-        )
+        cols_with_vocabs = list(self.categories.keys())
+        columns = [
+            list(c) if isinstance(c, tuple) else c for c in columns if c not in cols_with_vocabs
+        ]
+        dsk, key = _category_stats(ddf, self._create_fit_options_from_columns(columns))
         return Delayed(key, dsk)
 
     def fit_finalize(self, categories):
         for col in categories:
             self.categories[col] = categories[col]
 
     def clear(self):
-        self.categories = {}
+        self.categories = deepcopy(self.vocabs)
+
+    def process_vocabs(self, vocabs):
+        categories = {}
+
+        if _is_dataframe_object(vocabs):
+            fit_options = self._create_fit_options_from_columns(list(vocabs.columns))
+            base_path = os.path.join(self.out_path, fit_options.stat_name)
+            os.makedirs(base_path, exist_ok=True)
+            for col in list(vocabs.columns):
+                col_df = vocabs[[col]]
+                if col_df[col].iloc[0] is not None:
+                    if isinstance(col_df, pd.DataFrame):
+                        col_df = pd.DataFrame(
+                            {col: pd.concat([pd.Series([None]), col_df[col]]).reset_index()[0]}
+                        )
+                    else:
+                        import cudf
```

Comment:
@benfred Is this consistent with the way y'all are handling imports in the CPU-only changes?

Reply:
I think we can hide some of this complexity here. We have a 'concat' function in nvtabular.dispatch that does much the same here.

https://github.com/NVIDIA/NVTabular/blob/9f25d42932f107528882c91ccae77906b9f9502f/nvtabular/dispatch.py#L307-L311
 
I think we just need to add something to do the same as _make_df (https://github.com/NVIDIA/NVTabular/blob/9f25d42932f107528882c91ccae77906b9f9502f/nvtabular/dispatch.py#L314-L321) to make the ```pd.Series[None]``` or ```cudf.Series([None])``` for the initial value.